daffodil-core/src/main/scala/org/apache/daffodil/core/dsom/ElementBase.scala

```diff
@@ -774,10 +774,32 @@ trait ElementBase
     if (
       result.isDefined && repElement.isSimpleType && representation == Representation.Binary
     ) {
+      val nBits = result.get
+      if (isSignedIntegerType && nBits < 2) {
+        val outOfRangeStr =
+          "Minimum length for a signed binary integer is 2 bits, number of bits %d out of range. " +
+            "An unsigned integer with length 1 bit could be used instead."
+        if (tunable.allowSignedIntegerLength1Bit) {
+          SDW(
+            WarnID.SignedBinaryIntegerLength1Bit,
+            outOfRangeStr,
+            nBits
+          )
+        } else {
+          SDE(
+            outOfRangeStr,
+            nBits
+          )
+        }
+      }
+      schemaDefinitionWhen(
+        isUnsignedIntegerType && nBits < 1,
+        "Minimum length for an unsigned binary integer is 1 bit, number of bits %d out of range.",
+        nBits
+      )
       primType match {
         case primNumeric: NodeInfo.PrimType.PrimNumeric =>
           if (primNumeric.width.isDefined) {
-            val nBits = result.get
             val width = primNumeric.width.get
```

Comment:
Thoughts on renaming `width` to `maxWidth` and adding a new `minWidth` member in NodeInfo. Avoids some of the special logic around figuring out if stuff is signed or not. And then checking maxWidth and minWidth is very similar logic, just something like
```scala
if (primNumeric.minWdith.isDefined) {
   val minWidth = primNumeric.minWidth.get
   if(nBits < minWidth) {
     SDE(...)
   }
}
```
And then the minWidth can be reused in the parser/unparser logic.

Maybe we also add a `signed` member so we don't have to have the isSigned/isUnsigned stuff? And then parser/unparsers, which should already have access to the primType can just reference the signed member? We pass around isSigned in a bunch of places, wondering if maybe that can all be simplified.